From 8073053ca04ea8c9a62e16c882cea5b90f05b048 Mon Sep 17 00:00:00 2001
From: Sven Eckelmann <sven.eckelmann@gmx.de>
Date: Fri, 7 Aug 2009 21:45:23 +0200
Subject: [PATCH] Backport to 2.6.18

The function random32 was first introduced in v2.6.19-rc2-109-gaaa248f
were it was renamed from net_random.
The in-kernel socket api was first introduced in v2.6.18-310-gac5a488.
It is leightweight enough to implement needed functions as wrapper.

Signed-off-by: Sven Eckelmann <sven.eckelmann@gmx.de>
---
 compat.h |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/compat.h b/compat.h
index 0ea2655..53c1749 100644
--- a/compat.h
+++ b/compat.h
@@ -22,6 +22,15 @@
 
 #include <linux/version.h>	/* LINUX_VERSION_CODE */
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 19)
+
+#define random32() net_random()
+
+#define kernel_bind(_sock, _addr, _addrlen) \
+	_sock->ops->bind(_sock, _addr, _addrlen)
+
+#endif /* LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 19) */
+
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 20)
 
 #define delayed_work work_struct
-- 
1.6.3.3

